src/mono/mono/mini/intrinsics.c

```diff
@@ -878,7 +878,7 @@ mini_emit_inst_for_method (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSign
 			ins->klass = mono_defaults.runtimetype_class;
 			*ins_type_initialized = TRUE;
 			return ins;
-		} else if (!cfg->backend->emulate_mul_div && strcmp (cmethod->name, "InternalGetHashCode") == 0 && fsig->param_count == 1 && !mono_gc_is_moving ()) {
+		} else if (!cfg->backend->emulate_mul_div && (strcmp (cmethod->name, "InternalGetHashCode") == 0 || strcmp (cmethod->name, "InternalTryGetHashCode") == 0) && fsig->param_count == 1 && !mono_gc_is_moving ()) {
 			int dreg = alloc_ireg (cfg);
```

Comment:
Could this return 0 ?

Reply:
Note that we don't have a gc which has !mono_gc_is_moving () so this code is not really used right now.

Reply:
Oh, good point. The Boehm GC returns false for `mono_gc_is_moving`, so this needs to be addressed.

Reply:
Short term, it might be ok to just assert here. Alternatively, the caller of the icall could check for 0 and change it to 1.

Reply:
I took a closer look and both this and my changes to the interpreter are dead code. This is because both of them are expecting the `InternalGetHashCode` method to to be on `System.Object`, while the method were moved to `RuntimeHelpers` in 931c310bfd813f370de46dafb0ec16ba503ab14d .

My plan is to fix the interpreter and delete this code in `intrinsics.c`, since it is currently dead code and the Boehm GC does not seem to build in this repo anyways. Sound reasonable?

Reply:
Yes

Reply:
Done.